cmd/gen-go-sample/out.go

```diff
@@ -217,7 +230,8 @@ func writeLoop(l *schema_v1p2.LoopSpec, st *symTab, gen *generator) error {
 	p := gen.pt.Printf
 
 	sc, report := initScanner(l.Collection)
-	path, typ, err := writePath(sc, st, gen.descInfo)
+	path, itree, err := writePath(sc, st, gen.descInfo, false)
+	typ := itree.typ
```

Comment:
Can we put  the `typ` assignment after the error check? I understand that `itree` won't be `nil`, but for style sake lets keep the  error check immediately after the originating call 😄